LambdaEngine/Source/Game/ECS/Systems/Networking/ClientSystem.cpp

```diff
@@ -161,64 +172,59 @@ namespace LambdaEngine
 	{
 		UNREFERENCED_VARIABLE(pClient);
 
-		if (pPacket->GetType() == NetworkSegment::TYPE_ENTITY_CREATE)
+		auto iterator = m_PacketSubscribers.find(pPacket->GetType());
+		if (iterator != m_PacketSubscribers.end())
 		{
-			BinaryDecoder decoder(pPacket);
-			bool isMySelf		= decoder.ReadBool();
-			int32 networkUID	= decoder.ReadInt32();
-			glm::vec3 position	= decoder.ReadVec3();
-			glm::vec3 color		= decoder.ReadVec3();
-
-			if (isMySelf)
-				m_NetworkUID = networkUID;
-
-			Job addEntityJob;
-			addEntityJob.Components =
+			const TArray<std::function<void(NetworkSegment*)>>& functions = iterator->second;
+			for (const auto& func : functions)
 			{
-				{ RW, PositionComponent::Type()		},
-				{ RW, RotationComponent::Type()		},
-				{ RW, ScaleComponent::Type()		},
-				{ RW, MeshComponent::Type()			},
-				{ RW, NetworkComponent::Type()		},
-				{ RW, ControllableComponent::Type() }
-			};
-			addEntityJob.Function = std::bind(&ClientSystem::CreateEntity, this, networkUID, position, color);
-
-			ECSCore::GetInstance()->ScheduleJobASAP(addEntityJob);
+				func(pPacket);
+			}
 		}
-		else if (pPacket->GetType() == NetworkSegment::TYPE_PLAYER_ACTION)
+		else
 		{
-			ECSCore* pECS = ECSCore::GetInstance();
+			LOG_WARNING("No packet subscription of type: %hu", pPacket->GetType());
+		}
+	}
 
-			GameState serverGameState = {};
+	void ClientSystem::OnPacketCreateEntity(NetworkSegment* pPacket)
+	{
+		BinaryDecoder decoder(pPacket);
+		bool isMySelf		= decoder.ReadBool();
+		int32 networkUID	= decoder.ReadInt32();
+		glm::vec3 position	= decoder.ReadVec3();
+		glm::vec3 color		= decoder.ReadVec3();
 
-			BinaryDecoder decoder(pPacket);
-			int32 networkUID					= decoder.ReadInt32();
-			serverGameState.SimulationTick		= decoder.ReadInt32();//added
-			serverGameState.Position			= decoder.ReadVec3();
+		if (isMySelf)
+			m_NetworkUID = networkUID;
 
-			if (networkUID == m_NetworkUID)
-			{
-				std::scoped_lock<SpinLock> lock(m_Lock);
-				m_FramesProcessedByServer.PushBack(serverGameState);
-			}
-			else
-			{
-				auto pair = m_Entities.find(networkUID);
-				if (pair != m_Entities.end())
-				{
-					auto* pPositionComponents = pECS->GetComponentArray<PositionComponent>();
-
-					PositionComponent& positionComponent = pPositionComponents->GetData(pair->second);
-
-					positionComponent.Position	= serverGameState.Position;
-					positionComponent.Dirty		= true;
-				}
-				else
-				{
-					LOG_ERROR("NetworkUID: %d is not registred on Client", networkUID);
-				}
-			}
+		Job addEntityJob;
+		addEntityJob.Components =
+		{
+			{ RW, PositionComponent::Type()		},
+			{ RW, RotationComponent::Type()		},
+			{ RW, ScaleComponent::Type()		},
+			{ RW, MeshComponent::Type()			},
```

Comment:
Do you need MeshComponent here?
Do they all need to be RW, maybe just R or even NDA (No Data Access, just used for filtering)